awaitility/src/main/java/org/awaitility/core/JavaVersionDetector.java

```diff
@@ -4,18 +4,24 @@ public final class JavaVersionDetector {
     private static final String JAVA_VERSION = System.getProperty("java.version");
 
     public static int getJavaMajorVersion() {
+        return getJavaMajorVersion(JAVA_VERSION);
+    }
+
+    static int getJavaMajorVersion(String javaVersion) {
         final String normalizedJavaVersion;
-        if (JAVA_VERSION == null || JAVA_VERSION.isEmpty()) {
+        if (javaVersion == null || javaVersion.isEmpty()) {
             // Fallback to java 8
             normalizedJavaVersion = "8";
-        } else if (JAVA_VERSION.startsWith("1.")) {
-            normalizedJavaVersion = JAVA_VERSION.substring(2, 3);
+        } else if (javaVersion.startsWith("1.")) {
+            normalizedJavaVersion = javaVersion.substring(2, 3);
+        } else if (javaVersion.endsWith("-ea")) {
+            normalizedJavaVersion = javaVersion.substring(0, javaVersion.length() - 3);
```

Comment:
Temurin 23 EA builds have `java.version == 23-beta`. For example:

```
jdk-23+15/bin/java -XshowSettings:properties --version 2>&1 | grep java\\.version
    java.version = 23-beta
    java.version.date = 2024-09-17
```

This would need to be handled as well.

Reply:
Added a test for `-beta`, and replaced the explicit `-ea` handling with uniform handling for `-<anything>`.